compiler/src/iree/compiler/InputConversion/Common/Passes.td

```diff
@@ -33,6 +33,12 @@ def LinalgQuantizedMatmulToMatmulPass
   let constructor = "mlir::iree_compiler::InputConversion::createLinalgQuantizedMatmulToMatmulPass()";
 }
 
+def TensorOpsToLinalgPass
+    : InterfacePass<"iree-tensor-ops-to-linalg", "mlir::FunctionOpInterface"> {
+  let summary = "Convert tensor ops to linalg ops";
+  let constructor = "mlir::iree_compiler::InputConversion::createTensorOpsToLinalgPass()";
+}
```

Comment:
Both `tensor` and `linalg` are upstream dialects, so we'd prefer for a pass like this to exist upstream (https://github.com/llvm/llvm-project/tree/main/mlir/include/mlir/Conversion/TensorToLinalg, https://github.com/llvm/llvm-project/tree/main/mlir/lib/Conversion/TensorToLinalg). I also see that the `TensorOpsToLinalg.cpp` file only has `#include`s from `mlir/` and none from `iree/` (besides the pass definition header), so that's a good sign that the pass could be written upstream.

That being said, we do sometimes stage opinionated passes that may be more specialized to IREE here, with the goal of upstreaming them eventually. If this pass does end up living in IREE then we'd need to decide where in IREE, and I'm not sure if `InputConversion/Common` is the right place, would want others to weigh in 🤔 

As for how upstream would view such a pattern, I'm not sure. I think it would be welcome? https://discourse.llvm.org/t/rfc-adding-gather-scatter-ops/64757 goes into some detail on why the `tensor.gather` op should exist. What upstream conversions exist for that op today?

Reply:
thank you for the reply. So its probably best if I ask in the mlir forum if there is any use for this? As for "What upstream conversions exist for that op today?" I tried looking for anything that uses this op but I couldn't find anything but maybe I'm just searching for the wrong thing.

Reply:
I think its fine to have conversion from `tensor` to `linalg` in IREE. There is a case to be made to have this live upstream, but IREE uses Linalg as the main dialect for compilation, with some `tensor` operations to provide the glue between `Linalg` ops (like `tensor.extract/insert_slice` and `tensor.expand/collapse_shape`. So this is an IREE decision. If there is value in putting this upstream as a "common utility", then we can upstream it. If not, `compiler/InputConversion/Common` seems like a good place to add this AFAICS.